profiles/common/modules/features/nexteuropa_newsroom/classes/NewsroomHelper.inc

```diff
@@ -0,0 +1,633 @@
+<?php
+
+/**
+ * @file
+ * Code for the NewsroomHelper class.
+ */
+
+/**
+ * NewsroomHelper class.
+ */
+class NewsroomHelper {
+
+  /**
+   * Merges a term_tid and its children into one array.
+   *
+   * @param int $term_tid
+   *   Term id.
+   * @param int $term_vid
+   *   Vid id.
+   *
+   * @return array.
+   *   tids
+   */
+  public static function getTermChildren($term_tid, $term_vid = NULL) {
+    // We need parent ID also in the output array to use it in query.
+    $parent_children_tids = array($term_tid);
+    $children_tids = array_keys(taxonomy_get_children($term_tid, $term_vid));
+    if (count($children_tids) > 0) {
+      $parent_children_tids = array_merge($parent_children_tids, $children_tids);
+    }
+
+    return $parent_children_tids;
+  }
+
+  /**
+   * Helper function to serialize PDO object into an array.
+   */
+  public static function convertPdoToArray($query) {
+    $data = array();
+    if ($query) {
+      while ($result = $query->fetchAssoc()) {
+        $data[] = $result;
+      }
+    }
+
+    return $data;
+  }
+
+  /**
+   * Returns newsroom URL.
+   *
+   * @return string
+   *   Newsroom URL.
+   */
+  public static function getNewsroomUrl() {
+    return 'newsroom/' . implode('/', func_get_args());
+  }
+
+  /**
+   * Returns newsroom agenda URL.
+   *
+   * @return string
+   *   Newsroom agenda URL.
+   */
+  public static function getNewsroomAgendaUrl() {
+    return 'newsroom-agenda/' . implode('/', func_get_args());
+  }
+
+  /**
+   * Prepare options for the content type select.
+   *
+   * @param string $type
+   *   Type of the options to generate.
+   *
+   * @return array
+   *   Array of options.
+   */
+  public static function prepareOptions($type) {
+    $options = array();
+    if (is_numeric($type)) {
```

Comment:
I really, don't understand what is the goal of this comment.
1) if it is beauty of the code, then for me it is more clear to use this code
`
for ($i = 1; $i <= 100; $i++) {
    $options[$i] = $i;
  }
`
then this one
array_combine(range(1,$type),range(1,$type));

2) if it is code performance issue the loop would be faster then range function.
Results for 60000 items
Total: 0.017255067825317
Total: 0.024206161499023
but even for 5 items it is faster.
Total: 5.9604644775391E-6
Total: 1.4066696166992E-5

So, from wrote above we will keep the code as it is